src/material-experimental/mdc-slider/slider-thumb.ts

```diff
@@ -6,11 +6,38 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {NumberInput} from '@angular/cdk/coercion';
+import {coerceNumberProperty, NumberInput} from '@angular/cdk/coercion';
 import {DOCUMENT} from '@angular/common';
-import {Directive, ElementRef, EventEmitter, Inject, Input, Output} from '@angular/core';
+import {
+  AfterViewInit,
+  Directive,
+  ElementRef,
+  EventEmitter,
+  Inject,
+  InjectionToken,
+  Input,
+  Output,
+} from '@angular/core';
 import {Thumb} from '@material/slider';
-import {MatSlider} from './slider';
+
+/**
+ * This is a dummy interface that just contains the properties and methods of MatSlider that are
+ * used by MatSliderThumb. Rather than directly referencing MatSlider, we use this interface when
+ * defining MAT_SLIDER to avoid a circular dependency between MatSlider and MatSliderThumb.
+ */
+interface MatSlider {
```

Comment:
Shouldn't this be implemented by the slider?

Reply:
The interface?

Reply:
Yeah, otherwise the class and the interface can go out of sync.